### PR TITLE
[SPARK-31288][SQL] Remove `getRawTable` in `alterPartitions`

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This PR (SPARK-31288) removes `getRawTable` from `alterPartitions`.

### Why are the changes needed?
Reduce calls to Hive Metastore.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Exist test.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @gatorsmile

**Comment by dongjoon-hyun** [comment]
Could you explain about the performance gain in your environment in the PR description?

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please distinguish the performance **before** and **after** in the PR description.

**Review by gatorsmile — CHANGES_REQUESTED** [review]
@wangyum The best way to resolve your production perf issue is to maintain a metadata cache service, which can be shared by multiple spark sessions or clients. You can have a full control of the metadata and decide the life cycle.

**Comment by wangyum** [comment]
Thank you @dongjoon-hyun @gatorsmile

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala [hunk 1]
```diff
@@ -234,7 +235,7 @@ case class DropTableCommand(
       }
     }
 
-    if (isTempView || catalog.tableExists(tableName)) {
+    if (isTempView || isTableExist) {
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala]: Hm, @wangyum, when `isTempView` is `true`, `catalog.tableExists(tableName)` won't be called by short-circuiting. So, this isn't a clean improvement but it has trade-off. Can you clarify this trade-off as well?

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by gatorsmile** [comment]
thanks for removing the deadcode.

**Pull request closed without merging**